### ISSUE TITLE
False-alarm when constructing a pipeline with PEFT models

### ISSUE THREAD

**Issue by B-Step62** [huggingface/transformers#29395]
### System Info

Python 3.9.8
transformers == 4.37.2
peft==0.8.2

### Who can help?

@Narsil 

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

Creating a pipeline object with PEFT model like this.
```
import transformers
from peft import LoraConfig, TaskType, get_peft_model

base_model_id = "Elron/bleurt-tiny-512"
base_model = transformers.AutoModelForSequenceClassification.from_pretrained(base_model_id)
tokenizer = transformers.AutoTokenizer.from_pretrained(base_model_id)

peft_config = LoraConfig(
    task_type=TaskType.SEQ_CLS, inference_mode=False, r=8, lora_alpha=32, lora_dropout=0.1
)
peft_model = get_peft_model(base_model, peft_config)

pipeline = transformers.pipeline(task="text-classification", model=peft_model, tokenizer=tokenizer)
```

The last line triggers an error message:
```
The model 'PeftModelForSequenceClassification' is not supported for text-classification. Supported models are....
```

The error is raised from [here](https://github.com/huggingface/transformers/blob/main/src/transformers/pipelines/base.py#L1041-L1045) but is a false alarm, because the pipeline just works fine (also for other PEFT models/pipelines I've tested). 

A similar issue was raised in PEFT repository (https://github.com/huggingface/peft/issues/218) but got auto-closed. There was one workaround suggested by the community, that is creating a pipeline with base model first and mutate `pipeline.model` to the PEFT model, but it is hacky and bypasses any other checks. 

### Expected behavior

Supposedly PEFT models don't change model input/output interface, so as long as the base model is the one supported by the pipeline, users should not see the error. Even if this assumption is not always correct, the message should be a soft warning like `PEFT model might not work correctly with pipeline xxx` rather than being raised by `logger.error`, otherwise it is misleading for many models that work fine.

**Comment by ArthurZucker** [comment]
cc @Rocketknight1 !

**Comment by Rocketknight1** [comment]
@B-step62 I've opened a PR to fix this at #29517. Can you test that the problem is now fixed? You can install from the test branch with `pip install --upgrade git+https://github.com/huggingface/transformers.git@fix_peft_model_in_pipelines`

**Comment by ArthurZucker** [comment]
As the PR was close, a potential solution was given [here](https://github.com/huggingface/transformers/pull/29517#issuecomment-1990988026). TLDR this is not planned solution:
> first load the peft adapter with AutoModelxxx interface then pass that model to the pipeline or pass a valid path to a Peft model!
